scripts/powershell/Verify-And-Merge-PRs.ps1

```diff
@@ -0,0 +1,145 @@
+param(
+    [Parameter(Mandatory = $true)]
+    $AuthToken,
+
+    [Parameter(Mandatory = $true)]
+    $PRDataArtifactPath
+)
+
+$PRObjs = [System.Collections.ArrayList]::new()
+$MergedPRs = [System.Collections.ArrayList]::new()
+$ClosedPRs = [System.Collections.ArrayList]::new()
+$UnexpectedState = [System.Collections.ArrayList]::new()
+$OpenAndCleanPRs = [System.Collections.ArrayList]::new()
+$OpenButDirtyPRs = [System.Collections.ArrayList]::new()
+
+
+$GitHubAPIUrlBase = "https://api.github.com/repos/"
+$Headers = @{
+    Authorization = "bearer $AuthToken"
+}
+
+function Merge-PRs()
+{
+    if (($OpenButDirtyPRs.Count -gt 0) -or ($ClosedPRs.Count -gt 0) -or ($UnexpectedState.Count -gt 0))
+    {
+        Write-Output "The following PRs are either closed, or in an Un-mergeable state"
+        Write-Output "If Closed, Please investigate why a Sync PR was closed prematurely."
+        Write-Output "Ensure all checks are green and pr is approved. Merge them manually or run the sync pipeline again"
+
+        foreach ($obj in $OpenButDirtyPRs)
+        {
+            Write-Output "`thttps://github.com/$($obj.RepoOwner)/$($obj.RepoName)/pull/$($obj.PRNumber): State: Un-Mergable"
+        }
+
+        foreach ($obj in $ClosedPRs)
+        {
+            Write-Output "`thttps://github.com/$($obj.RepoOwner)/$($obj.RepoName)/pull/$($obj.PRNumber): State: Closed"
+        }
+
+        foreach ($obj in $UnexpectedState)
+        {
+            Write-Output "`thttps://github.com/$($obj.RepoOwner)/$($obj.RepoName)/pull/$($obj.PRNumber): State: Unexpected"
+        }
+        exit 1
+    }
+
+    Write-Output "Will merge the Following PRs"
+    foreach ($obj in $OpenAndCleanPRs)
+    {
+        $MergeAPIUrl = "${GitHubAPIUrlBase}$($obj.RepoOwner)/$($obj.RepoName)/pulls/$($obj.PRNumber)/merge"
+        $CommitsAPIUrl = "${GitHubAPIUrlBase}$($obj.RepoOwner)/$($obj.RepoName)/pulls/$($obj.PRNumber)/commits"
+
+        $data = @{
+            commit_title = ""
+            commit_message = ""
+            sha = $obj.HeadSHA
+            merge_method = "squash"
+        }
+
+        # Get latest Commit of the pull request
+        try 
+        {
+            $response = Invoke-RestMethod -Method Get -Headers $Headers $CommitsAPIUrl
+            $data.commit_message = $response[$response.Count - 1].commit.message # last commit message of the pull request
```

Comment:
Is there a way to automatically have github squash the commit_message for the merge? What happens if you don't set them to anything?

Reply:
Yh, omitting  commit_message and commit_title does not affect merging the PR. It uses the default commit message.